### DISCUSSION TITLE
Suggestion: Add option to toggle spatial sound on playback device

### DISCUSSION THREAD

**Discussion by alessiocpt** [File-New-Project/EarTrumpet#660] [Ideas]
I sometimes enable spatial sound while watching movies or playing games with headphones. Is it possible to add a button to toggle it on and off? It could be visible only on right click on a playback device.

Thanks a lot for your work!

**Comment by Zoneykid** [comment]
Would like to request this as well since now windows has 3 Spatial Audio renderers (Dolby Atmos, DTS:X, and Windows Sonic) and the original volume context menu has a nice easy sub-menu to switch between them.

**Comment by red9350** [comment]
+1, I'd also like to have Windows' Spatial sound context menu replicated in EarTrumpet, so I could finally hide forever the System sound tray icon.

![image](https://user-images.githubusercontent.com/16651290/74430140-98753f80-4e5c-11ea-9f04-26ee6a9295eb.png)

**Reply by MiLeX84** [reply]
I would love to have this as an option in the eartrumpet menu indeed.

**Reply by red9350** [reply]
So, Windows 11 REMOVED this feature, and buried the toggle deep inside the Settings app. Now EarTrumpet is our only hope to quickly toggle Spatial sound

**Reply by Radder124** [reply]
Another reason to stay on W10 :) But yeah it would be great to have this feature back with Eartrumpet!

**Comment by mcbexx** [comment]
Add another one to the request pile. With Windows Sonic being free it's a very common feature to use when gaming with headphones. Being able to quickly switch it on and off would be great.

Bonus points if it would switch the device in question back to 5.1 when you turn Windows Sonic off again. I have no idea why Windows would not behave like that by default.

**Comment by howardsun-dev** [comment]
Hopefully, this will be implemented

**Comment by DaedalusBLN** [comment]
+1

Having a quickly accessible Spatial sound toggle is a crucial for me. As a Dolby Atmos for Headphones user, when using VLC to playback certain multichannel audio media files, I've had EXTREMELY LOUD bleeps coming out of my speaker instead of the file to be decoded normally. Switching DAfH off was the only way for these files to be played normally, so having quick access from the tray would make my life easier. Another usecase is when doing audio production work, you certainly want these effects to be off before digging in your DAW.

That being said, I don't know whether Microsoft is making it easy or possible at all to third-party devs to access/control these settings. Even the Dolby Access can't toggle itself and gives you instructions instead. I was messing around with AutoHotkey and just wanted to make a shortcut into my audio interface's Properties page (didn't even try to make it work for the currently default/active one). I failed!

Thanks for all your efforts so far, gave you 5* on Windows Store, well deserved!

**Comment by Radder124** [comment]
Do we have an update on this?

**Comment by doctor-nefario** [comment]
+1 also really requesting to add this feature.

**Comment by siddarththiru** [comment]
+1 I indeed use this very often too!!!

I badly want this feature to be implemented to I can stash away the default speaker icon to hidden menu

**Comment by Greyze** [comment]
The only reason the I still have the volume taskbar icon visible is because I often have to switch to spatial sound. Would be fantastic if eartrumpet added this feature.

+1

**Comment by AverageComet250** [comment]
I have the same issue as everyone here. I switch between wired earphones, laptop speakers, a wireless headset and a wired headset, and windows doesn't always save the config for spatial sound so i have to manually set it most of the time. It's my _only_ use for built in taskbar icon.

+4 (that's one for everyone in my family cause we all use eartrumpet)

**Comment by MiLeX84** [comment]
I fully support the idea of the spatial sound option in the EarTrumpet menu. 
This is currently the only feature missing, for now, we either have to have the windows sound shortcut as well, or go to windows legacy -> playback devices, select the device, go to properties, and then spatial sound. The shortcut in the original windows 10 sound menu is great and a major miss for EarTrumpet.

**Comment by miguedevel** [comment]
This is even more relevant now that Windows 11 doesn't have the right-click menu to switch Speaker channels/spatial outputs.

**Comment by ghost** [comment]
made a request before I saw this thread, pasting it here instead:

The Windows 10 native systray is inferior to EarTrumpet, however, it does feature a handy toggle for Spatial Audio:
![image](https://user-images.githubusercontent.com/17074394/128265544-07db6c49-0813-414d-a6dd-c0d2e6ee5f74.png)
Handy toggle in native systray:
![image](https://user-images.githubusercontent.com/17074394/128265417-8954056a-5668-4206-8d72-2d1ee076b991.png)

It would be cool if EarTrumpet had an easy on/off toggle (and hotkey option) for it:
![image](https://user-images.githubusercontent.com/17074394/128265781-bb17aaeb-e1ed-428f-a633-6ea19c416814.png)

**Reply by riverar** [reply]
Yep, thanks. We've gotten a bunch of these requests.

Windows (and third-party) Audio Processing Objects (APOs) are a little tricky to support but we hear the need for this.

**Comment by AD2011** [comment]
+1
